Header design

dutchmartin opened the pull request:
Start with recreating the design from #15
Depends on #2

Commit:
Start a header design

Kingdutch commented on src/components/atoms/ActionBotton.js:
```diff
@@ -0,0 +1,19 @@
+
+import styled from "styled-components";
+
+const Button = styled.div`
```
```suggestion
const ActionButton = styled.a`
  display: inline-block;
```

Making this an anchor tag allows the button to be exported as is without creating a separate wrapping React element.

Kingdutch commented on src/components/organisems/Header.js:
```diff
@@ -26,8 +29,13 @@ const Title = styled.div`
+   &:visited {
+    color: #fff;
+    text-decoration: none;
+   }
```
Can be simplified to:

```
margin: 0 5px;
&, &:visited {
  color: inherit;
  text-decoration: none;
}
```

This way changing the color in the header will also change the link color

Kingdutch commented on src/components/organisems/Header.js:
```diff
@@ -1,10 +1,13 @@
-  background-color: rgba(200,216,217,.3);
+  background-color: #211a74;
+  color: #fff;
+  display: flex;
```
I should've added comments but the `HeaderWrapper` is for the full-width background color with the `HeaderContainer` having the centered header content. So this wrapper probably doesn't need `display: flex;`

dutchmartin replied:
I am not sure what to do with it, but I use the display flex to align the button in the correct place.

Kingdutch commented on src/components/organisems/Header.js:
```diff
@@ -45,6 +53,7 @@ export default function Header({ siteTitle, isFront }) {
           <HeaderLink className={"focus:underline hover:underline"} to="/about">About</HeaderLink>
         </NavBar>
       </HeaderContainer>
+        <ActionBotton/>
```
Moving this inside of the `NavBar` component will probably put it where you want it. 

There's also a typo in the component and filename: `ActionBotton` -> `ActionButton`.

Moving the `href` and `text` here: `<ActionButton href="csp.rustfest.eu">Submit Talk</ActionButton>` will allow us to re-use the button in other places.

Kingdutch reviewed (commented):
Cool stuff! I've added some comments on where we can make improvements 😃

Commit:
Make the button layout pretty

Commit:
Fix the typo in Button

Commit:
Make the button reuseable

Commit:
Delete package-lock.json

Kingdutch commented:
Rebased this based on the changes in `feature/team-page`

Kingdutch commented:
I moved the Call to Action inside of the navbar so that it properly sits within the page and doesn't float all the way to the right on large screens. I've also changed the spacing a bit so that adding more links will work properly.

<img width="1438" alt="image" src="https://user-images.githubusercontent.com/327697/73660051-bcd05f80-4697-11ea-80d5-0eedfea556cc.png">

Kingdutch committed:
Move CallToAction into navbar

This ensures it doesn't drift outside of the page

Kingdutch committed:
Add hover state to call to action button

Kingdutch committed:
Remove React wrapper from styled-component

Styled Components are already React elements so there's usually no need
to wrap them in another React element. Not wrapping them also makes it
easier to utilise all the underlying dom element's properties.

Kingdutch committed:
Move logo to header

This prepares the header for use with different page content.
